stage0/run.go

```diff
@@ -539,27 +573,41 @@ func overlayRender(cfg RunConfig, img types.Hash, cdir string, dest string, appN
 	}
 
 	destRootfs := path.Join(dest, "rootfs")
-	if err := os.MkdirAll(destRootfs, 0755); err != nil {
+	if err := os.MkdirAll(destRootfs, defaultRegularDirPerm); err != nil {
 		return err
 	}
 
 	cachedTreePath := cfg.Store.GetTreeStoreRootFS(img.String())
 
-	overlayDir := path.Join(cdir, "overlay", img.String())
-	if err := os.MkdirAll(overlayDir, 0755); err != nil {
+	overlayDir := path.Join(cdir, "overlay")
+	if err := os.MkdirAll(overlayDir, defaultRegularDirPerm); err != nil {
+		return err
+	}
+
+	// Since the parent directory($POD_UUID) has the 'S_ISGID' bit, here
+	// we need to explicitly turn the bit off when creating this overlay
+	// directory so that it won't inherit the bit. Otherwise the files
+	// created by users within the pod will inherit the 'S_ISGID' bit
+	// as well.
+	if err := os.Chmod(overlayDir, defaultRegularDirPerm); err != nil {
+		return err
+	}
+
+	imgDir := path.Join(overlayDir, img.String())
+	if err := os.MkdirAll(imgDir, defaultRegularDirPerm); err != nil {
```

Comment:
Sometimes we read `/var/lib/rkt/pods/run/$POD_UUID/overlay/$IMAGE_ID/upper/rkt/status/$APP` so we'd need `/var/lib/rkt/pods/run/$POD_UUID/overlay/$IMAGE_ID` to be readable by users in the `rkt` group.